### ISSUE TITLE
Crash using typeof(Self) in OnCompile TypeDone

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1240]
Code to reproduce:
```
class Foo
{
    [Comptime, OnCompile(.TypeDone)]
    static void Apply()
    {
        let t = typeof(Self);
    }
}
```
Tested with: (Nightly 11/18/2021)

**Comment by hermansimensen** [comment]
Was just about to post this myself!
`FATAL ERROR
typeInstance->mDefineState != BfTypeDefineState_DefinedAndMethodsSlotted in C:\Jenkins\workspace\Beef\IDEHelper\Compiler\BfModuleTypeUtils.cpp:4509
`

**Comment by bfiete** [comment]
Fixed at f1105a8ce15761bfd5290ddb38035e83019c2542
